### ISSUE TITLE
inference.py but for models trained on NDDS-produced datasets

### ISSUE THREAD

**Issue by monajalal** [NVlabs/Deep_Object_Pose#277]
Do you have an inference.py also for dope/scripts/train.py where we train an item for NDDS-generated dataset?

I do see that you have an inference.py for dope/scripts/train2/inference.py which is for NVISII.

When I initially fed my .pth trained weights to it, gave me error and it could be it is designed for nvisii data?

It is rather weird though since using the inference.py inside train2/inference.py I see the created belief maps which makes quite a lot of sense however, it doesn't produce the json files for the test images.

```
len(all_peaks[-1]):  2
nb_object:  0
all_peaks[-1][nb_object][2]:  0.00017739832
nb_object:  1
all_peaks[-1][nb_object][2]:  0.017452825


config_inference/config_pose.yaml
117:thresh_points: 0.1

inference/detector.py
666:            if all_peaks[-1][nb_object][2] > config.thresh_points:
694:                    if candidate[2] < config.thresh_points:
```

Basically, here, `all_peaks[-1][nb_object][2]` is smaller than 0.1 which is thresh_points.

So to sum up, 
1. is there a separate inference.py for pth trained via dope/scripts/train.py or should we use the same file as in dope/scripts/train2/inference.py?
2. If I have to use dope/scripts/train2/inference.py could you please guide me on fixing or debugging the error?

**Comment by mintar** [comment]
To answer the first part of your question: no, it does not matter how you got your .pth file (train or train2, NDDS or NViSII). All .pth files are compatible with everything.

--
Sent from my phone. Short. Typos.

Am 09.12.2022 22:43 schrieb Mona Jalal ***@***.***>:

Do you have an inference.py also for dope/scripts/train.py where we train an item for NDDS-generated dataset?

I do see that you have an inference.py for dope/scripts/train2/inference.py which is for NVISII.

When I initially fed my .pth trained weights to it, gave me error and it could be it is designed for nvisii data?

It is rather weird though since using the inference.py inside train2/inference.py I see the created belief maps which makes quite a lot of sense however, it doesn't produce the json files for the test images.

len(all_peaks[-1]):  2
nb_object:  0
all_peaks[-1][nb_object][2]:  0.00017739832
nb_object:  1
all_peaks[-1][nb_object][2]:  0.017452825


config_inference/config_pose.yaml
117:thresh_points: 0.1

inference/detector.py
666:            if all_peaks[-1][nb_object][2] > config.thresh_points:
694:                    if candidate[2] < config.thresh_points:


Basically, here, all_peaks[-1][nb_object][2] is smaller than 0.1 which is thresh_points.

So to sum up,

  1.  is there a separate inference.py for pth trained via dope/scripts/train.py or should we use the same file as in dope/scripts/train2/inference.py?
  2.  If I have to use dope/scripts/train2/inference.py could you please guide me on fixing or debugging the error?
